core/baseline-core-v1.0-psd01.md

```diff
@@ -2786,113 +2795,729 @@ The document defines storage capability requirements in the following areas:
+8. Repeat steps 2-7 for each type of policy created in step 1.
+9. Verify that the BPI's user interface or API provides a clear way to manage policies.
+
+Test Passing Criteria: The test will pass if all of the following criteria are met:
+
+* All policies created in step 2 are successfully created, modified, and deleted, and are stored in the BPI storage component.
+* Retrieval of policies in step 6 returns the correct policy for each policy created in step 2.
+* Enforcement of policies in step 7 results in the appropriate actions for each policy created in step 2.
+* The BPI's user interface or API provides an intuitive way to manage policies.
+* There are no errors or exceptions during the test.
+
 #### **[R285]**	
 BPI Storage MUST support Single-Sign-On (SSO).
 
 *See [[SSO](#sso)] also for the recommendations of the National Institute of Standards and Technology (NIST Guide to Secure Web Services).*
 
+[[R285]](#r285) Testability: 
+
+Preconditions:
+
+* A BPI test environment is installed and configured properly.
+* The BPI storage component is designed to support SSO using a SSO provider.
+* A BPI Subject has appropriate permissions to access the BPI.
+
+Test Steps:
+
+1. Attempt to access the BPI's user interface or API without providing any authentication credentials.
+2. Verify that the BPI redirects the BPI Subject to a SSO login page or displays a message indicating that authentication is required.
+3. Log in using a SSO provider that has been configured for the BPI.
+4. Verify that the BPI grants access to the BPI Subject after successful authentication.
+5. Attempt to access a protected resource or perform a protected action within the BPI.
+6. Verify that the BPI grants access to the BPI Subject if the BPI Subject has appropriate permissions.
+7. Attempt to access a protected resource or perform a protected action within the BPI using an invalid or expired SSO token.
+8. Verify that the BPI denies access to the BPI Subject and provides an appropriate error message.
+9. Attempt to log out of the BPI.
+10. Verify that the BPI logs out the BPI Subject from the SSO provider and revokes the BPI Subject's BPI session.
+
+Test Passing Criteria: The test will pass if
+
+* The BPI redirects the BPI Subject to a SSO login page or displays a message indicating that authentication is required.
+* Successful login using a SSO provider grants access to the BPI.
+* Access to protected resources or actions within the BPI is granted only to BPI Subjects with appropriate permissions.
+* Invalid or expired SSO tokens result in access being denied and an appropriate error message being displayed.
+* Logging out of the BPI logs the BPI Subject out of the SSO provider and revokes the BPI Subject's BPI session.
+
+
 #### **[R286]**	
 BPI Storage MUST support multi-factor authentication (MFA)
 
 *See the link here for the NIST definition adopted in this document [MFA](https://csrc.nist.gov/glossary/term/mfa).*
```

Comment:
```suggestion
*This document adopts [NIST's definition of MFA](https://csrc.nist.gov/glossary/term/mfa).*
```

Reply:
Which one, given that there are two listed there?

Reply:
they are all functionally equivalent.